cmd/notary/main_test.go

```diff
@@ -162,8 +167,203 @@ func TestInsufficientArgumentsReturnsErrorAndPrintsUsage(t *testing.T) {
 			[]string{"-c", filepath.Join(tempdir, "idonotexist.json"), "-d", tempdir}, arglist...))
 		err = cmd.Execute()
 
-		require.NotContains(t, err.Error(), "error opening config file", "running %s", invalid)
+		require.NotContains(t, err.Error(), "error opening config file", "running `notary %s`", invalid)
 		// it's a usage error, so the usage is printed
-		require.Contains(t, b.String(), "Usage:", "expected usage when running %s", invalid)
+		require.Contains(t, b.String(), "Usage:", "expected usage when running `notary %s`", invalid)
 	}
 }
+
+// The bare notary command and bare subcommands all print out usage
+func TestBareCommandPrintsUsageAndNoError(t *testing.T) {
+	tempdir, err := ioutil.TempDir("", "empty-dir")
+	require.NoError(t, err)
+	defer os.RemoveAll(tempdir)
+
+	// just the notary command
+	b := new(bytes.Buffer)
+	cmd := NewNotaryCommand()
+	cmd.SetOutput(b)
+
+	cmd.SetArgs([]string{"-c", filepath.Join(tempdir, "idonotexist.json")})
+	require.NoError(t, cmd.Execute(), "Expected no error from a help request")
+	// usage is printed
+	require.Contains(t, b.String(), "Usage:", "expected usage when running `notary`")
+
+	// notary key, notary cert, and notary delegation
+	for _, bareCommand := range []string{"key", "cert", "delegation"} {
+		b := new(bytes.Buffer)
+		cmd := NewNotaryCommand()
+		cmd.SetOutput(b)
+
+		cmd.SetArgs([]string{"-c", filepath.Join(tempdir, "idonotexist.json"), bareCommand})
+		require.NoError(t, cmd.Execute(), "Expected no error from a help request")
+		// usage is printed
+		require.Contains(t, b.String(), "Usage:", "expected usage when running `notary %s`", bareCommand)
+	}
+}
+
+type recordingMetaStore struct {
+	gotten []string
+	storage.MemStorage
+}
+
+// GetCurrent records what metadata was requested
```

Comment:
nit: maybe the comment should mention that it also gets the metadata from the underlying metastore? (same for `GetChecksum`)